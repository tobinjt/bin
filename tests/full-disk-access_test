#!/bin/bash

set -e -f -u -o pipefail

DIR="$(mktemp -d -t full-disk-access-XXXXXXXXXX)"
STDOUT="${DIR}/stdout"
STDERR="${DIR}/stderr"
readonly DIR STDOUT STDERR

cc -Weverything -o "${DIR}/full-disk-access" "${HOME}/bin/full-disk-access.c"

"${DIR}/full-disk-access" echo foo >> "${STDOUT}" 2>> "${STDERR}"
"${DIR}/full-disk-access" does-not-exist >> "${STDOUT}" 2>> "${STDERR}" || true
diff -u full-disk-access_test.stdout "${STDOUT}"
diff -u full-disk-access_test.stderr "${STDERR}"
