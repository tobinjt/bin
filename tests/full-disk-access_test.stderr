Executing "does-not-exist" failed: No such file or directory
